### ISSUE TITLE
Using `meta` as a channel name in a DSL2 workflow is forbidden

### ISSUE THREAD

**Issue by Midnighter** [nextflow-io/nextflow#2246]
### Expected behavior and actual behavior

I wanted to use a channel named `meta` in one of my workflows. This is not possible and clashes with some internals of nextflow or groovy it seems. I did not see this documented anywhere so my apologies if this is not a bug.

### Steps to reproduce the problem

1. A definition similar to this one produces below output error.

```groovy
#!/usr/bin/env nextflow

nextflow.enable.dsl = 2

process FOO {
  input:
  val(meta)

  """
  touch "${meta.id}"
  """
}


workflow {
  meta = Channel.of([id: 'sequence1'])
  FOO(meta)
}
```

2. When the channel is defined outside the workflow, there is a different but similar error.

```groovy
meta = Channel.of([id: 'sequence1'])

workflow {
  FOO(meta)
}
```

### Program output 

1.
```
Cannot cast object 'DataflowBroadcast around DataflowStream[?]' with class 'groovyx.gpars.dataflow.DataflowBroadcast' to class 'nextflow.script.ScriptMeta'
```

2.
```
Error executing process > 'FOO'

Caused by:
  No such property: id for class: nextflow.script.ScriptMeta

Source block:
  """
    touch "${meta.id}"
    """
```

### Environment 

* Nextflow version: 21.04.3 build 5560 
* Java version: Groovy 3.0.7 on OpenJDK 64-Bit Server VM 11.0.11+9-Ubuntu-0ubuntu2.20.04
* Operating system: Linux 5.11.0-7620-generic
* Bash version: GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)

**Comment by abhi18av** [comment]
Hi @Midnighter ,

I'm no Groovy expert but it seems that the word `meta` is used a lot within the Groovy DSL constructs therefore I think that it is a part of Area 51 of Nextflow (a Groovy DSL). 

If you'd like to use `meta` identifier in a different way, then perhaps a usage example could be seen in `nf-core/modules` repository via [this search](https://github.com/nf-core/modules/search?q=meta).

**Comment by Midnighter** [comment]
Thanks for your pointers. I'm not obsessed with the name, no. It's rather that I couldn't find any reference in the documentation on this. Just a warning somewhere that says, don't name your channel `meta` as that is an internal keyword would be enough for me.

**Comment by pditommaso** [comment]
Well spotted, actually `meta` is an internal `private` data structure, as such should not be accessible, not sure why this is happening. A possible workaround consist defining meta in the local scope eg. `def meta = ..`

**Comment by Midnighter** [comment]
Right, the mysteries of using or not using `def` 😆 I can confirm that it works without error when using `def meta` either inside or outside the workflow definition. However, `Channel(...).set { meta }` again does not. It creates an error like 2. above.

**Comment by pditommaso** [comment]
Investigating for a patch

**Comment by pditommaso** [comment]
Solved by ceceb0784
